tests/test_big_modeling.py

```diff
@@ -325,6 +337,24 @@ def test_dispatch_model_multi_gpu(self):
             output = model(x)
             self.assertTrue(torch.allclose(expected, output.cpu(), atol=1e-5))
 
+    @require_cuda
+    def test_dispatch_model_copy(self):
+        original_model = ModelForTestCopy(id=1)
+        device_map = {"linear1": 0, "batchnorm": "cpu", "linear2": 0}
+
+        x = torch.randn(2, 3)
+        expected, original_output_id = original_model(x)
+
+        dispatch_model(original_model, device_map)
+
+        copied_model = copy.deepcopy(original_model)
+        copied_model.id = 2
+        output, copied_output_id = copied_model(x)
+
+        self.assertEqual(original_model.id, original_output_id)
+        self.assertEqual(copied_model.id, copied_output_id)
```

Comment:
Maybe we can add a test that check if the forward is not the same 
```suggestion
        self.assertEqual(copied_model.id, copied_output_id)
        self.assertFalse(copied_model.linear1.forward is original_model.linear1.forward)
```

Reply:
Added!